ref/general/requestTiming.adoc

```diff
@@ -0,0 +1,71 @@
+:page-layout: general-reference
+:page-type: general
+:page-description: The request timing feature automatically detects slow requests and logs the information when the request is completed.
+:page-categories: MicroProfile
+:seo-title: The request timing feature automatically detects slow and hung requests
+:seo-description: The request timing feature automatically detects slow and hung requests and logs the information when the request is completed.
+= Slow and hung request detection
+
+When an application takes longer than expected to complete a request, the request can be logged as slow or hung. Your applications need a continuous monitoring tool to detect and analyze slow and hung requests.
+The Request Timing feature automatically detects slow  and hung requests and logs the information when the request is completed.
+
+The Request Timing feature monitors requests with respect to time and automatically detects and diagnoses slow and hung requests, and also logs when the request is completed.
+The feature also logs the information about a completed slow request with the request id and time of completion.
+Also, if a previously hung request is completed, a detailed message is logged, with the ID and time of completion of the request.
+
+== Configure slow and hung request detection
+
+You can enable the Request Timing feature by adding `requestTiming`-`1.0` to the list of features in the `server.xml` file.
+
+[source,java]
+----
+<featureManager>
+      ...
+      <feature>requestTiming-1.0</feature>
+      ...
+</featureManager>
+----
+
+The slow and hung request thresholds are configured by setting the `slowRequestThreshold` and `hungRequestThreshold` attributes on the `requestTiming` element in your `server.xml` file.
+These attributes specify global slow and hung request thresholds for all requests that are timed.
+
+== Detect slow requests
+
+When a request runs for longer than the `slowRequestThreshold` or any configured threshold, the Request Timing feature generates a warning message in the messages log file.
+The message includes details about the request and events that made up the request.
+The default threshold value is 10 seconds.
+
+The following example shows the format of a log messages.
+
+[source,java]
+----
+TRAS0112W: Request <(Request ID)> has been running on thread <THREADID> for at least <DURATION>. The following stack trace shows what this thread is currently running.
+<STACK TRACE>
+<DURATION AND OPERATIONS Table>
+----
+
+The message helps you analyze the reason behind the slow request, though it is difficult to determine whether the request is hung at that point or is still running slowly.
+To address this issue, three messages are displayed for any slow request at the interval of specified `<slowRequestThreshold>`.
```

Comment:
change
at the interval of specified `<slowRequestThreshold>`.
to
at the interval specified by the `<slowRequestThreshold>` attribute.